File: backend/src/main/java/online/nonamelab/WorkSite_Work/site/model/Site.java
package online.nonamelab.WorkSite_Work.site.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import online.nonamelab.WorkSite_Work.common.entiry.BaseEntity;
import online.nonamelab.WorkSite_Work.user.model.User;

@Entity
@Table(name = "sites")
@Getter
@Setter
@NoArgsConstructor
public class Site extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private String description;

    // LOCATION INFO
    private String country;
    private String city;
    private String address;

    // Optional GPS (future check-in / tracking)
    private Double latitude;
    private Double longitude;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private SiteStatus status;

    // RESPONSIBLE MANAGER
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "manager_id")
    private User manager;

}

File: frontend/src/types/site.ts
export type SiteStatus = "PLANNED" | "ACTIVE" | "PAUSED" | "INACTIVE" | "ARCHIVED";

export type SiteResponse = {
  id: number;
  name: string;
  description: string | null;
  country: string;
  city: string;
  address: string;
  status: SiteStatus;
  managerId: number;
  managerFullName: string;
  createdAt: string;
};
